### PR TITLE
✨ migrate history store to pinia

### PR DESCRIPTION
**Pull request by roiLeo**
## PR Type

- [x] Refactoring

## Context

- [x] related #4755 
- [x] Remove unused components

### CODE REVIEW

File: stores/history.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        (nft) =>
+          !isThisMonth(new Date(nft.date)) &&
+          !isToday(new Date(nft.date)) &&
+          !isYesterday(new Date(nft.date))
+      ),
+    getCurrentlyViewedItem: (state) => state.currentlyViewedItem,
+    getCurrentlyViewedCollection: (state) => state.currentlyViewedCollection,
+  },
+  actions: {
+    addHistoryItem(payload) {
```
Comment (vikiival) [inline stores/history.ts:94]: Never seen so many if-else

**Review by daiagi — APPROVED** [review]

File: stores/history.ts (hunk 1 above)
Comment (roiLeo) [reply stores/history.ts:94]: 😭 I won't refactor this lol

File: stores/history.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  }),
+  getters: {
+    getVisitedToday: (state) =>
+      state.visitedNFTs.filter((nft) => isToday(new Date(nft.date))),
+    getVisitedYesterday: (state) =>
+      state.visitedNFTs.filter((nft) => isYesterday(new Date(nft.date))),
+    getVisitedPastWeek: (state) =>
+      state.visitedNFTs.filter(
+        (nft) =>
+          isThisWeek(new Date(nft.date)) &&
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply stores/history.ts:71]: I feel you

File: stores/history.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        if (this.visitedNFTs.length > 30) {
+          this.visitedNFTs.pop()
+        }
+        this.visitedNFTs.unshift(historyItem)
+      }
+    },
+    setCurrentlyViewedCollection(payload) {
+      this.currentlyViewedCollection = payload
+    },
+    removeHistoryItem(payload) {
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply stores/history.ts]: 👋 bye bye

File: components/shared/Sharing.vue [hunk 4]
```diff
@@ -119,7 +120,11 @@ export default class Sharing extends Vue {
 
   private active = false
 
-  private hashtags = 'KusamaNetwork,KodaDot'
+  public hashtags = 'KusamaNetwork,KodaDot'
+
+  get historyStore() {
+    return useHistoryStore()
+  }
```
Comment (Jarsen136) [inline components/shared/Sharing.vue:127]:
```suggestion
  private historyStore = useHistoryStore()
```
I guess there's no need to repeatedly call the getter function every time we access the store.

Comment (roiLeo) [reply components/shared/Sharing.vue:127]: hmmm... does it work? didn't tried like that

Comment (Jarsen136) [reply components/shared/Sharing.vue]: Yes, it would work. Let's give it a shot

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 50 usd

**Pull request merged**